Rails31

in4systems opened the pull request:
Hi,

I've made several changes to the adapter. It has mostly been a case of fixing bugs when I come across them. There are some new features, like the ability to do "rake db:reset".

I would like the changes to be merged, so more people get the benefit.

Chris

Commit:
Fix deprecation warning about Arel::Visitors.

Commit:
Rails create (database insert) will mostly work.

I think it's meant to return something, which I don't currently do.
It shouldn't print so much.
It probably shouldn't commit after every single insert.
It should handle date columns using the column override class.

Commit:
Support dropping tables.

Support rake db:drop.
Unlike most databases systems, this doesn't delete the database. Creating a
SQLAnywhere database requires a lot of configuration. Therefore it is easier to simply
blank out the database on db:drop.

Commit:
Move the removal of quotes around column default to the SQL.

Now that I'm looking at the changes, I realize this probably wasn't needed.

Commit:
Fix primary_key

Commit:
Fix the adapter to run through our seed file.

Binary and decimal binds need to be handled specially.

Commit:
Create the last_inserted_id method.

Commit:
Fix Model.first.

The changes are in lib/arel/visitors/sqlanywhere.rb and sqlanywhere_adapter.rb:455

Also, indented.

Commit:
Fix id_value error

Commit:
Make the adapter tolerant against changing capitalization only in column names.

Commit:
Change how the primary key is calculated.

Commit:
Fix primary key method

Commit:
Fix joining tables.

Before this change, the join part of the SQL statement would be missing.

Commit:
Fix binding date types.

Commit:
Actually fix how Arel::Visitors is included.

Better error reporting.
Most uses of SQLAnywhereInterface check their return values.
Errors no longer silently fail.

Commit:
Change the remove_column method to take an array of columns to drop.

This is what active records expects.

Commit:
Fix handling of offsets.

Pagination would fail everytime before hand.

Commit:
handle numeric data types.

Previously, you might get a numeric as the result of a calculation involving decimal types.
Previously, this would be converted to a ruby string type.
Now, it is converted to BigDecimal.

Commit:
Added purge task (required for testing).

Use bit for boolean (as it can be null).

Commit:
Make bit columns nullable by defualt.

Commit:
Remove the :limit => 4 for integer columns in the schema.rb file

Commit:
Fix deleting multiple columns.

Sometimes, column_names would be an array inside an array.

Commit:
Revert changes where I used bit instead of tinyint for boolean.

Tinyint now works for boolean (which it didn't previously).

Commit:
Handle nulls of type decimal.

Also, remove redundant line.

Commit:
bump version

Commit:
Change top (x) to top x.

It's strange that I wasn't getting the error on Thriftwood

Commit:
allow connection_string in the database.yml.

Fix binding of nils. (nil Integers were being saved as 0).

Attempt to fix problem with aggregate calculations on groups.

clst commented:
Thank you very much. Our company is currenly moving so I will not have time to look at your changes right away but I will certainly merge them asap.

Commit:
Fixed the ActiveRecord::Result does not support -@ error.
This was due to rails expecting to have the number of affected rows after a delete, but instead having a (empty?) result set.

Commit:
Sqlanywhere 9 does not like the character in varchar (255 character).

Commit:
New version of Active record was producing an error.

The to_sql method was trying to access @visitor but finding nil.
Does this fix it?

Commit:
Find the major version of SQLAnywhere.

Don't run the indexes of SQLAnywhere earlier than 11.

clst commented:
Thanks, it also seems to work fine on Rails 3.2.3

Great work fixing this mess ;)
